src/components/Blocks/Header.tsx

```diff
@@ -0,0 +1,180 @@
+    } else {
+      updatedSearchParams.delete('search');
+    }
+
+    setSearchParams(updatedSearchParams);
+  }, [searchQuery]);
+
+  return (
+    <header className="header">
+      <Link to="/" className="header__home-link home-link" />
+
+      <div className="header__wrapper">
+        <nav
+          className="header__nav nav"
+          style={isShoppingCartActive() as React.CSSProperties}
+        >
+          <ul className="nav__list">
+            <li className="nav__item">
+              <NavLink
+                className={({ isActive }) => navLinkClassName({ isActive })}
+                to="/"
+              >
+                Home
+              </NavLink>
+            </li>
+
+            <li className="nav__item">
+              <NavLink
+                className={({ isActive }) => navLinkClassName({ isActive })}
+                to="/phones"
+              >
+                Phones
+              </NavLink>
+            </li>
+
+            <li className="nav__item">
+              <NavLink
+                className={({ isActive }) => navLinkClassName({ isActive })}
+                to="/tablets"
+              >
+                Tablets
+              </NavLink>
+            </li>
+
+            <li className="nav__item">
+              <NavLink
+                className={({ isActive }) => navLinkClassName({ isActive })}
+                to="/accessories"
+              >
+                Accessories
+              </NavLink>
+            </li>
+          </ul>
+        </nav>
+
+        <section className="header__header-bar">
+          <form
+            className="header__search"
+            onSubmit={(event) => event.preventDefault()}
+            style={isSearchVisible() as React.CSSProperties}
+          >
+            <input
+              type="search"
+              placeholder={`Search in ${currentProduct}...`}
+              className="header__search--bar"
+              onChange={event => setSearchQuery(event.target.value)}
+            />
+
+            {searchQuery.length > 0
+            && (
+              <button
+                type="submit"
+                className="header__search--button cancel-search"
+                onClick={() => setSearchQuery('')}
+                data-cy="searchDelete"
+              />
+            )}
+
+            <button
+              type="submit"
+              className="header__search--button search-products"
+            />
+          </form>
+          <NavLink
+            to="/favorites"
+            className={({ isActive }) => `header__favorites ${shoppingIconClassName({ isActive })}`}
+            style={isShoppingCartActive() as React.CSSProperties}
+          >
+            {totalAmountOfFavorites > 0
+            && (
+              <p
+                className="shopping-icon__amount"
+              >
+                {totalAmountOfFavorites}
+              </p>
+            )}
+          </NavLink>
+          <NavLink
+            to="/shopping-cart"
```

Comment:
```suggestion
            to="/cart"
```
just to fix [tests](https://github.com/mate-academy/react_phone-catalog/actions/runs/6014641731/job/16314804333?pr=211)